riscv/triggers.cc

```diff
@@ -148,6 +148,59 @@ match_result_t mcontrol_t::memory_access_match(processor_t * const proc, operati
   return MATCH_NONE;
 }
 
+reg_t itrigger_t::tdata1_read(const processor_t * const proc) const noexcept
+{
+  auto xlen = proc->get_xlen();
+  reg_t tdata1 = 0;
+  tdata1 = set_field(tdata1, CSR_ITRIGGER_TYPE(xlen), CSR_TDATA1_TYPE_ITRIGGER);
+  tdata1 = set_field(tdata1, CSR_ITRIGGER_DMODE(xlen), dmode);
+  tdata1 = set_field(tdata1, CSR_ITRIGGER_HIT(xlen), hit);
+  tdata1 = set_field(tdata1, CSR_ITRIGGER_VS, vs);
+  tdata1 = set_field(tdata1, CSR_ITRIGGER_VU, vu);
+  tdata1 = set_field(tdata1, CSR_ITRIGGER_NMI, nmi);
+  tdata1 = set_field(tdata1, CSR_ITRIGGER_M, m);
+  tdata1 = set_field(tdata1, CSR_ITRIGGER_S, s);
+  tdata1 = set_field(tdata1, CSR_ITRIGGER_U, u);
+  tdata1 = set_field(tdata1, CSR_ITRIGGER_ACTION, action);
+  return tdata1;
+}
+
+void itrigger_t::tdata1_write(processor_t * const proc, const reg_t val) noexcept
+{
+  auto xlen = proc->get_xlen();
+  assert(get_field(val, CSR_ITRIGGER_TYPE(xlen)) == 4);
+  dmode = proc->get_state()->debug_mode ? get_field(val, CSR_ITRIGGER_DMODE(xlen)) : 0;
+  hit = get_field(val, CSR_ITRIGGER_HIT(xlen));
+  vs = proc->extension_enabled('H') ? get_field(val, CSR_ITRIGGER_VS) : 0;
+  vu = proc->extension_enabled('H') ? get_field(val, CSR_ITRIGGER_VU) : 0;
```

Comment:
This is a bit fishy...if `misa.H` is changed from 1 to 0, some previously-set triggers could still have `vs` and `vu` set, and that would be visible via `csrr tdata1`.

Would it be sufficient to do the masking only in `tdata1_read()` based on the current `misa.H`?